packages/algob/src/lib/tx.ts

```diff
@@ -149,31 +186,80 @@ async function sendAndWait (
-
-    let txns = [];
-    for (const t of execParams) { txns.push(await mkTx(t)); }
-    txns = algosdk.assignGroupID(txns);
-    signedTxn = txns.map((txn: Transaction, index: number) => {
-      const signed = signTransaction(txn, execParams[index]);
-      deployer.log(`Signed transaction ${index}`, signed);
-      return signed;
-    });
-  } else {
-    const txn = await mkTx(execParams);
-    signedTxn = signTransaction(txn, execParams);
-    deployer.log(`Signed transaction:`, signedTxn);
+  try {
+    let signedTxn;
+    let txns: Transaction[] = [];
+    const txIDMap = new Map<number, string>();
+    if (Array.isArray(execParams)) {
+      if (execParams.length > 16) { throw new Error("Maximum size of an atomic transfer group is 16"); }
+
+      for (const [index, t] of execParams.entries()) {
+        txns.push(await mkTx(deployer, t, index, txIDMap));
+      }
+
+      txns = algosdk.assignGroupID(txns);
+      signedTxn = txns.map((txn: Transaction, index: number) => {
+        const signed = signTransaction(txn, execParams[index]);
+        deployer.log(`Signed transaction ${index}`, signed);
+        return signed;
+      });
+    } else {
+      const txn = await mkTx(deployer, execParams, 0, txIDMap);
+      signedTxn = signTransaction(txn, execParams);
+      deployer.log(`Signed transaction:`, signedTxn);
+      txns = [txn];
+    }
+    const confirmedTx = await sendAndWait(deployer, signedTxn);
+    console.log(confirmedTx);
+    await registerCheckpoints(deployer, txns, txIDMap);
+    return confirmedTx;
+  } catch (error) {
+    deployer.storeCheckpoint();
+
+    console.log(error);
+    throw error;
+  }
+}
+
+/**
+ * Register checkpoints for ASA and SSC
+ * @param deployer Deployer object
+ * @param txns transaction array
+ * @param txIDMap transaction map index to name
+ */
+async function registerCheckpoints (
+  deployer: Deployer,
+  txns: Transaction[],
+  txIDMap: Map<number, string>
+): Promise<void> {
+  for (const [i, txn] of txns.entries()) {
+    let txConfirmation;
+    const name = txIDMap.get(i);
+    switch (txn.type) {
+      case 'acfg': {
+        txConfirmation = await deployer.waitForConfirmation(txn.txID());
+        const asaInfo: ASAInfo = {
+          creator: encodeAddress(txn.from.publicKey),
+          txId: txn.txID(),
+          assetIndex: txConfirmation["asset-index"],
+          confirmedRound: txConfirmation['confirmed-round']
+        };
+        if (name) deployer.registerASAInfo(name, asaInfo);
+        break;
+      }
+      case 'appl': {
+        txConfirmation = await deployer.waitForConfirmation(txn.txID());
+        const sscInfo: SSCInfo = {
+          creator: encodeAddress(txn.from.publicKey),
+          txId: txn.txID(),
+          appID: txConfirmation['application-index'],
+          confirmedRound: txConfirmation['confirmed-round']
+        };
+        if (name) deployer.registerSSCInfo(name, sscInfo);
+        break;
+      }
+    }
```

Comment:
I had a question. In `deployer.deployASA(..)` & `deployer.deploySSC(..)` we are also pushing *txWriter* logs (using `deployer.txWriter.push(..)` ).

Are we doing it here as well?

Reply:
no, we are not, thanks for catching will add it here